python/demo/gmsh/demo_gmsh.py

```diff
@@ -5,33 +5,101 @@
 # SPDX-License-Identifier:    LGPL-3.0-or-later
 #
 # =====================================
-# Mesh generation using Gmsh and pygmsh
+# Mesh generation using the GMSH python API
 # =====================================
 
-import numpy as np
-import pygmsh
+from dolfinx.mesh import create_mesh, create_meshtags
+from dolfinx.io import XDMFFile, ufl_mesh_from_gmsh
+from dolfinx.cpp.io import perm_gmsh, extract_local_entities
+from dolfinx import cpp
 from mpi4py import MPI
+import gmsh
+import numpy as np
+
+
+def gather_gmsh_physical_entities():
+    """
+    Convenience function that extracts all entitites with
+    physical tags, creating mesh data structures for each
+    cell type.
+    Returns a dictionary where the key is the gmsh cell type
+    and its content is a dictionary of cells and cell data.
+
+    """
+    phys_grps = gmsh.model.getPhysicalGroups()
+    meshes = {}
+    for dim, tag in phys_grps:
+        entities = gmsh.model.getEntitiesForPhysicalGroup(dim, tag)
+        for entity in entities:
+            element_types, elem_tags, node_tags =\
+                gmsh.model.mesh.getElements(dim, tag=entity)
+            for (e_type, e_tags, n_tags) in zip(
+                    element_types, elem_tags, node_tags):
+                name, dim, order, num_nodes, local_coords, _ = \
+                    gmsh.model.mesh.getElementProperties(e_type)
+                e_cells = n_tags.reshape(-1, num_nodes) - 1
+                if e_type in meshes.keys():
+                    meshes[e_type]["cells"] =\
+                        np.concatenate((meshes[e_type]["cells"],
+                                        e_cells), axis=0)
+                    meshes[e_type]["cell_data"] = \
+                        np.concatenate((meshes[e_type]["cell_data"],
+                                        np.full(len(e_tags), tag)), axis=0)
+                else:
+                    meshes[e_type] = {"cells":
+                                      e_cells,
+                                      "cell_data": np.full(len(e_tags), tag)}
+
+    return meshes
+
+
+def sort_gmsh_points():
```

Comment:
The is confusing without an in argument. Or the docstring is unclear.

Reply:
There are no in-arguments as everything is stored in the `gmsh.model`.
I will try to improve the doc string.